Improve kubectl-crossplane cmd logging and add verbose flag

psinghal20 opened the pull request:
### Description of your changes
This change fixes some of the silent failures in the `kubectl crossplane` cmd and adds a `--verbose` flag for verbose logging.

Fixes #2237 

Please let me know if any of the statements needs rewording or if you find some statements unnecessary for the CLI.

I have:

- [x] Read and followed Crossplane's [contribution process].
- [x] Run `make reviewable test` to ensure this PR is ready for review.
- [ ] Added `backport release-x.y` labels to auto-backport this PR if necessary.

### How has this code been tested
I have tested this PR locally primarily and update the supplied build test for this change.

[contribution process]: https://git.io/fj2m9

hasheddan commented on cmd/crank/build.go:
```diff
@@ -46,49 +47,62 @@ type buildCmd struct {
 	defer func() { _ = f.Close() }()
-	return tarball.Write(nil, img, f)
+	if err := tarball.Write(nil, img, f); err != nil {
+		logger.Info("Failed to write package image", "error", err)
```
I think we will want to return error here :)

psinghal20 replied:
Done. PTAL.

hasheddan commented on cmd/crank/update.go:
```diff
@@ -75,16 +97,32 @@ type updateProviderCmd struct {
 	if err != nil {
-		return errors.Wrap(warnIfNotFound(err), "cannot update provider")
+		err = warnIfNotFound(err)
+		logger.Info("Failed to update configuration", "error", err)
```
```suggestion
		logger.Info("Failed to update provider", "error", err)
```

hasheddan commented on cmd/crank/update.go:
```diff
@@ -75,16 +97,32 @@ type updateProviderCmd struct {
+		logger.Info("Failed to update configuration", "error", err)
+		return errors.Wrap(err, "cannot update provider")
 	}
+	logger.Info("Found previous configuration object")
```
```suggestion
	logger.Info("Found previous provider object")
```

hasheddan commented on cmd/crank/update.go:
```diff
@@ -75,16 +97,32 @@ type updateProviderCmd struct {
 	if err != nil {
-		return errors.Wrap(warnIfNotFound(err), "cannot update provider")
+		err = warnIfNotFound(err)
+		logger.Info("Failed to update configuration", "error", err)
```
```suggestion
		logger.Info("Failed to update provider", "error", err)
```

hasheddan commented on cmd/crank/update.go:
```diff
@@ -95,11 +133,15 @@ func (c *updateProviderCmd) Run(k *kong.Context) error { // nolint:gocyclo
 	if err != nil {
-		return errors.Wrap(warnIfNotFound(err), "cannot update provider")
+		err = warnIfNotFound(err)
+		logger.Info("Failed to update configuration", "error", err)
```
```suggestion
		logger.Info("Failed to update provider", "error", err)
```

hasheddan commented on cmd/crank/update.go:
```diff
@@ -95,11 +133,15 @@ func (c *updateProviderCmd) Run(k *kong.Context) error { // nolint:gocyclo
 	if err != nil {
-		return errors.Wrap(warnIfNotFound(err), "cannot update provider")
+		err = warnIfNotFound(err)
+		logger.Info("Failed to update configuration", "error", err)
```
```suggestion
		logger.Info("Failed to update provider", "error", err)
```

hasheddan commented on cmd/crank/update.go:
```diff
@@ -38,16 +39,33 @@ type updateConfigCmd struct {
+	logger.Info("Found kube config")
+	kube, err := typedclient.NewForConfig(kubeConfig)
+	if err != nil {
+		logger.Info(errKubeClient, "errors", err)
```
I notice you are using the key `errors` in some places and `error` in others, is there a specific reason for each?

hasheddan commented on cmd/crank/install.go:
```diff
@@ -75,10 +78,12 @@ func (c *installConfigCmd) Run(k *kong.Context) error {
 	if pkgName == "" {
 		ref, err := name.ParseReference(c.Package)
 		if err != nil {
+			logger.Info(errPkgIdentifier, "error", err)
```
It won't have a huge impact right now because we are passing a `NopLogger` if not `--verbose`, but it would likely make more sense for all of these to be `logger.Debug` rather than `logger.Info`. Some good pointers [here](https://crossplane.io/docs/v1.2/contributing/observability_developer_guide.html#in-practice).

psinghal20 commented:
Hi @hasheddan, I have fixed all the typos(There were a lot, sorry for that :sweat_smile: )
I have changed the logging level to debug too, thanks for the feedback!

hasheddan commented on cmd/crank/install.go:
```diff
@@ -42,6 +43,8 @@ import (
 
 const (
 	errPkgIdentifier = "invalid package image identifier"
+	errKubeConfig    = "failed to get kube-config"
```
```suggestion
	errKubeConfig    = "failed to get kubeconfig"
```

Nitpick: can you change all references to `kubeconfig` to be one word? :)

psinghal20 replied:
Done!

hasheddan replied:
@psinghal20 I think you may have missed changing it in this error message

hasheddan reviewed (approved):
Thanks @psinghal20! One nitpick then I think we should be good to merge :)
